pkg/prom/instance/group_manager.go

```diff
@@ -227,7 +231,19 @@ func (m *GroupManager) deleteConfig(name string) error {
 }
 
 // Stop stops the Manager and all of its managed instances.
-func (m *GroupManager) Stop() { m.inner.Stop() }
+func (m *GroupManager) Stop() {
+	m.mtx.Lock()
+	defer m.mtx.Unlock()
+
+	for group := range m.groups {
+		if err := m.inner.DeleteConfig(group); err != nil {
+			level.Error(m.log).Log("msg", "failed to delete group", "name", group, "err", err)
```

Comment:
Good question, and it depends on what the error is. I'd have to double check (I'm not at home) but I think the only error this can return right now is if the group doesn't exist anymore, but in the future if it ever reports a problem for anything else, it means there are scrape jobs running that failed to be removed.

Reply:
I double-checked: this will only happen if the config was already deleted. We shouldn't see this log in practice.